Off-by-one in assertion error when retrieving constraints

The assertion in `get_constrs` here: https://github.com/JuliaOpt/Gurobi.jl/blob/64f071fb65b837fd65f7847504875d1f5b0d5b27/src/grb_constrs.jl#L168 is off by one. When the model has no constraints, `start = 1`, but `m = 0`, which causes the assertion to fail. For example: 

```julia
julia> model = Gurobi.Model(Gurobi.Env(), "model")
Academic license - for non-commercial use only
Gurobi Model: model
    type   : LP
    sense  : minimize
    number of variables             = 0
    number of linear constraints    = 0
    number of quadratic constraints = 0
    number of sos constraints       = 0
    number of non-zero coeffs       = 0
    number of non-zero qp objective terms  = 0
    number of non-zero qp constraint terms = 0


julia> Gurobi.get_constrmatrix(model)
ERROR: AssertionError: start <= m
Stacktrace:
 [1] get_constrs(::Gurobi.Model, ::Int64, ::Int64) at /home/rdeits/locomotion/explorations/simpleqp-mpc/packages/v0.6/Gurobi/src/grb_constrs.jl:168
 [2] get_constrmatrix(::Gurobi.Model) at /home/rdeits/locomotion/explorations/simpleqp-mpc/packages/v0.6/Gurobi/src/grb_constrs.jl:163
```

I'll submit a fix.

--- a/src/grb_constrs.jl
+++ b/src/grb_constrs.jl
@@ -165,8 +165,9 @@ end
 function get_constrs(model::Model, start::Integer, len::Integer)
 
     m = num_constrs(model)
-    @assert start <= m
-    @assert len <= m
+    @assert start >= 1
+    @assert len >= 0
+    @assert start + len <= m + 1
     n = num_vars(model)
     numnzP = Array{Cint}(1)
     cbeg = Array{Cint}(len+1)

--- a/test/runtests.jl
+++ b/test/runtests.jl
@@ -35,7 +35,16 @@ end
 include("constraint_modification.jl")
 
 @testset "Empty constraints (Issue #142)" begin
-    model = Gurobi.Model(Gurobi.Env(), "model")
-    A = Gurobi.get_constrmatrix(model)
-    @test size(A) == (0, 0)
+    @testset "No variables, no constraints" begin
+        model = Gurobi.Model(Gurobi.Env(), "model")
+        A = Gurobi.get_constrmatrix(model)
+        @test size(A) == (0, 0)
+    end
+    @testset "One variable, no constraints" begin
+        model = Gurobi.Model(Gurobi.Env(), "model")
+        Gurobi.add_cvar!(model, 0.0)
+        Gurobi.update_model!(model)
+        A = Gurobi.get_constrmatrix(model)
+        @test size(A) == (0, 1)
+    end
 end

--- a/test/runtests.jl
+++ b/test/runtests.jl
@@ -33,3 +33,9 @@ end
 end
 
 include("constraint_modification.jl")
+
+@testset "Empty constraints (Issue #142)" begin
+    model = Gurobi.Model(Gurobi.Env(), "model")
+    A = Gurobi.get_constrmatrix(model)
+    @test size(A) == (0, 0)
+end

